### PR TITLE
Switch agent storage from .kibana  to .fleet-agents 

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

* Switch agent storage from .kibana saved objects to .fleet-agents index
* Improve generated Go structs, common Unmarshall func for all ESDocument backed structures
* Remove ActionDoc, switch to model.Action.
* Improve the data layer package.
* Generalize the hits unmarshal code.
* Switch agent authentication from saved object to .fleet-agents index
* Remove legacy EnrollmentApiKey structure
* Update actions dispatcher to use model.Action
* Remove feature flag config

@blakerouse I know you are working on the policy stuff, I tried not to touch it much. Need to catch up on Monday on who is doing what next.

### CODE REVIEW

**Comment by ruflin** [comment]
@nchaulet For your awarness.

File: internal/pkg/dl/unmarshal.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+package dl
+
+import (
+	"encoding/json"
+	"fleet/internal/pkg/es"
+	"fleet/internal/pkg/model"
+)
+
+func Unmarshal(hit es.HitT, v interface{}) error {
```
Comment (blakerouse) [inline internal/pkg/dl/unmarshal.go:13]: Why not put this on the actual type that way `json.Unmarshal` will do it for you?

```
func (hit *es.HitT) Unmarshal(data []byte) error {
   ..unmarshal..
}
```

File: internal/pkg/model/schema.go [hunk 2]
```diff
@@ -13,8 +13,28 @@ import (
...
+	ESInitialize(id string, seqno, version int64)
+}
+
+type ESDocument struct {
+	Id      string `json:"-"`
+	Version int64  `json:"-"`
+	SeqNo   int64  `json:"-"`
+}
+
+func (d *ESDocument) ESInitialize(id string, seqno, version int64) {
```
Comment (blakerouse) [inline internal/pkg/model/schema.go:26]: This is great!

**Review by blakerouse — APPROVED** [review]
Looks really good.

I like the removal of the features, the hard switch to the index. Just one question about the `Unmarshal` but otherwise looks good.

File: internal/pkg/dl/unmarshal.go (hunk 1 above)
Comment (aleksmaus) [reply internal/pkg/dl/unmarshal.go:13]: Sure, will update

**Pull request merged**